pkgs/development/libraries/gstreamer/bad/default.nix

```diff
@@ -42,18 +43,27 @@ stdenv.mkDerivation rec {
   #       old version of Apple SDK's.
   configureFlags = optional stdenv.isDarwin "--disable-cocoa";
 
-  patchPhase = ''
+  preConfigure = ''
+    patchShebangs .
+  '';
+
+  prePatch = ''
     sed -i 's/openjpeg-2.2/openjpeg-${openJpegVersion}/' ext/openjpeg/*
   '';
 
+  patches = [
+    ./fix_install_dir.patch
+    ./fix_pkgconfig_includedir.patch
+  ];
+
   src = fetchurl {
     url = "${meta.homepage}/src/gst-plugins-bad/${name}.tar.xz";
-    sha256 = "1v5z3i5ha20gmbb3r9dwsaaspv5fm1jfzlzwlzqx1gjj31v5kl1n";
+    sha256 = "17sgzgx1c54k5rzz7ljyz3is0n7yj56k74vv05h8z1gjnsnjnppd";
   };
 
   outputs = [ "out" "dev" ];
 
-  nativeBuildInputs = [ pkgconfig python ];
+  nativeBuildInputs = [ meson ninja pkgconfig python ];
```

Comment:
`gettext` is missing.

Reply:
Why isn't `gettext` and `ninja` automatically propagated by meson? But I will add it.

Reply:
Not every project uses i18n. And ninja is not needed for meson to work, you can use other builders.